composites/Plugin/Shared/components/Toggle.js

```diff
@@ -0,0 +1,135 @@
+import PropTypes from "prop-types";
+import React from "react";
+import styled from "styled-components";
+import colors from "../../../../style-guide/colors.json";
+import { __ } from "@wordpress/i18n";
+
+const ToggleBar = styled.div`
+	background-color: ${ props => props.isEnabled ? "#a5d6a7" : colors.$color_button_border };
+	border-radius: 7px;
+	height: 14px;
+	width: 30px;
+	cursor: pointer;
+	margin: 0;
+	outline: 0;
+	&:focus > span {
+		box-shadow: inset 0 0 0 1px ${colors.$color_white}, 0 0 0 1px #5b9dd9, 0 0 2px 1px rgba(30, 140, 190, .8);
+	}
+`;
+
+const ToggleBullet = styled.span`
+	background-color: ${ props => props.isEnabled ? colors.$color_green_medium_light : colors.$color_grey_medium_dark };
+	margin-left: ${ props => props.isEnabled ? "12px" : "-2px" };
+	box-shadow: 0 2px 2px 2px rgba(0, 0, 0, 0.1);
+	border-radius: 100%;
+	height: 20px;
+	width: 20px;
+	position: absolute;
+	margin-top: -3px;
+`;
+
+const ToggleVisualLabel = styled.span`
+	font-size: 14px;
+	line-height: 20px;
+	width: 30px;
+	text-align: center;
+	display: inline-block;
+	margin: 0;
+	font-style: italic;
+`;
+
+const ToggleDiv = styled.div`
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+`;
+
+class Toggle extends React.Component {
+	/**
+	 * Sets the toggle object.
+	 *
+	 * @param {Object} props The props to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( props ) {
+		super( props );
+
+		this.onClick = this.props.onToggleDisabled;
+
+		this.setToggleState = this.setToggleState.bind( this );
+
+		if ( props.disable !== true ) {
+			this.onClick = this.setToggleState.bind( this );
+		}
+	}
+
+	/**
+	 * Returns the rendered HTML.
+	 *
+	 * @returns {ReactElement} The rendered HTML.
+	 */
+	render() {
+		return(
+			<ToggleDiv>
+				{ /* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions */ }
+				<label
+					htmlFor={ this.props.id }
```

Comment:
Not important enough for now, and also more like a preference than an instruction. 

but when props are used in multiple places, it's nice to extract props to local variables, this is done, for example, in SnippetEditorFields.js.